### PR TITLE
48682 - Update the VAOS FE code to use Drupal Source of Truth for Cerner information

### LINKED ISSUE #48682: Update the VAOS FE code (va_online_scheduling_use_dsot) - IsCerner changes
## Description
Public Website team plans to deprecate the [file](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/utilities/cerner/index.js) that our `isCerner` flag references. VAOS FE code uses the `isCerner` flag to determine if the Veteran is associated with a VAMC facility that uses Cerner or Vista. 

When the facility uses Cerner (the name of the facility is listed in the file), the Veteran will be routed to the Cerner portal from the VAOS static homepage. When the facility uses Vista, the Veteran will be directed to VAOS to book or request an appointment. 

We completed the research on the [spike ticket ](https://app.zenhub.com/workspaces/vaos-team-603fdef281af6500110a1691/issues/department-of-veterans-affairs/va.gov-team/47253) and will need to complete the work and tested before December 1, 2022. 

---

## Dev Notes
The changes will be behind the 'va_online_scheduling_use_dsot' feature flag. 
The function, 'connectDrupalSourceOfTruthCerner' will be called when creating a new appointment. 

This function call will dispatch the following redux events:

FETCH_STATIC_DATA_STARTED
FETCH_STATIC_DATA_SUCCEEDED
NOTE: FETCH_STATIC_DATA_SUCCEEDED depends on the existence of the file, vamc-ehr.json. This file is generated during the Drupal content build.

You can generate the file during development or retrieved the file from production (https://www.va.gov/data/cms/vamc-ehr.json). The file should be placed in the 'build/localhost/data/cms' directory. Use the following command when running the content build locally:

yarn build --pull-drupal --use-cached-assets
This can take a wile to complete so I retrieved the file from production.

The changes required to use the Drupal driven Cerner API are in the following files:

../vaos/routes.jsx - Dispatch the redux event when the new appointment page is loaded
../vaos/redux/selectors.js - Determine which function to call depending on feature flag
The bulk of the changes consists of determining which function to call depending on the feature flag, 'vaOnlineSchedulingAcheronService' being true or false. Please see the following branch for details.

[department-of-veterans-affairs/vets-website#22489](https://github.com/department-of-veterans-affairs/vets-website/pull/22489)
NOTE: Please ignore the other changes that fix the eslint cyclic dependencies warnings

---

## Acceptance Criteria
Background: the `va_online_scheduling_use_dsot` toggle is on

- [ ] Cerner file isn't being referenced any longer.
- [ ] When the facility uses Cerner, the Veteran will be routed to the Cerner portal from the VAOS static homepage.
- [ ]  When the facility uses Vista, the Veteran will be directed to VAOS to book or request an appointment.

### Design Assets
None

## Definition of Done
- [ ] All acceptance criteria are met
- [ ] Documentation is updated

---

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
This PR implements changes to use the Drupal as the source of truth for determining if a veteran is a Cerner user.

**How to test**

- Turn on feature flag - vaOnlineSchedulingUseDsot
- Modify ../vaos/services/mocks/index.js:409 to set facility id = 442
- Schedule an appointment
- Verify link to route user to Cerner exist on 'Choose a VA location' page

**NOTE**: 'is_cerner' flag in index.js is not used since Cerner information is coming from Drupal Source of Truth. Also, real facility ids are used.

## Original issue(s)
fixes department-of-veterans-affairs/va.gov-team#48682

## Testing done
Unit testing

## Screenshots
localhost_3001_health-care_schedule-view-va-appointments_appointments_ (4)

## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/routes.jsx [hunk 1]
```diff
@@ -35,6 +36,7 @@ export default function createRoutesWithStore(store) {
             component={asyncLoader(() =>
               import(/* webpackChunkName: "vaos-form" */ './new-appointment')
                 .then(({ NewAppointment, reducer }) => {
+                  connectDrupalSourceOfTruthCerner(store.dispatch);
```
Comment (vbahinwillit) [inline src/applications/vaos/routes.jsx:39]: Dispatch redux event to get Cerner information.

File: src/applications/vaos/services/appointment/index.js [hunk 2]
```diff
@@ -4,7 +4,7 @@
  */
 import moment from 'moment-timezone';
 import * as Sentry from '@sentry/browser';
-import environment from 'platform/utilities/environment';
+import environment from '@department-of-veterans-affairs/platform-utilities/environment';
 import recordEvent from 'platform/monitoring/record-event';
```
Comment (vbahinwillit) [inline src/applications/vaos/services/appointment/index.js:8]: Fix eslint warnings.

File: src/applications/vaos/services/appointment/index.js [hunk 3]
```diff
@@ -51,33 +51,7 @@ import {
...
-  'INPATIENT/CHECKED IN',
-  'INPATIENT/CHECKED OUT',
-  'INPATIENT/FUTURE',
-  'INPATIENT/NO ACT TAKN',
-  'NO ACTION TAKEN',
-  'NO-SHOW & AUTO RE-BOOK',
-  'NO-SHOW',
-  'NON-COUNT',
-]);
+import { getProviderName } from '../../utils/appointment';
```
Comment (vbahinwillit) [inline src/applications/vaos/services/appointment/index.js:54]: Moved to fix cyclic dependency eslint warning.

File: src/applications/vaos/services/appointment/index.js [hunk 4]
```diff
@@ -51,33 +51,7 @@ import {
...
-  'INPATIENT/ACT REQ',
-  'INPATIENT/CHECKED IN',
-  'INPATIENT/CHECKED OUT',
-  'INPATIENT/FUTURE',
-  'INPATIENT/NO ACT TAKN',
-  'NO ACTION TAKEN',
-  'NO-SHOW & AUTO RE-BOOK',
-  'NO-SHOW',
-  'NON-COUNT',
-]);
```
Comment (vbahinwillit) [inline src/applications/vaos/services/appointment/index.js:80]: Moved to fix eslint cyclic dependency warning.

File: src/applications/vaos/services/appointment/index.js [hunk 5]
```diff
@@ -666,22 +640,6 @@ export function isVideoHome(appointment) {
...
-export function getPatientInstruction(appointment) {
-  if (appointment?.patientInstruction.includes('Medication Review')) {
-    return 'Medication Review';
-  }
-  if (appointment?.patientInstruction.includes('Video Visit Preparation')) {
-    return 'Video Visit Preparation';
-  }
-  return null;
-}
-
```
Comment (vbahinwillit) [inline src/applications/vaos/services/appointment/index.js:684]: Moved to fix eslint cyclic dependency warning.

File: src/applications/vaos/services/appointment/index.js [hunk 6]
```diff
@@ -929,32 +888,6 @@ export async function cancelAppointment({
...
-      .filter(person => !!person.name)
-      .map(person => `${person.name.given.join(' ')} ${person.name.family} `);
-    if (providers.length > 0) {
-      return providers;
-    }
-    return null;
-  }
-  return null;
-}
-
```
Comment (vbahinwillit) [inline src/applications/vaos/services/appointment/index.js:957]: Moved to fix eslint cyclic dependency warning.

File: src/applications/vaos/services/appointment/transformers.js [hunk 7]
```diff
@@ -201,7 +200,7 @@ function getAppointmentDuration(appt) {
       appt.vvsAppointments?.[0]?.duration,
     10,
   );
-  return isNaN(appointmentLength) ? 60 : appointmentLength;
+  return Number.isNaN(appointmentLength) ? 60 : appointmentLength;
```
Comment (vbahinwillit) [inline src/applications/vaos/services/appointment/transformers.js:203]: Fix eslint warning

File: src/applications/vaos/services/location/index.js [hunk 8]
```diff
@@ -264,22 +263,6 @@ export function getSiteIdFromFacilityId(id) {
...
- */
-export function getTestFacilityId(facilityId) {
-  if (!environment.isProduction() && facilityId) {
-    return facilityId.replace('442', '983').replace('552', '984');
-  }
-
-  return facilityId;
-}
-
 /**
```
Comment (vbahinwillit) [inline src/applications/vaos/services/location/index.js:266]: Moved to fix eslint cyclic dependency warning.

File: src/applications/vaos/services/location/transformers.js [hunk 9]
```diff
@@ -53,14 +53,10 @@ function isFacilityOpenAllDay(hours) {
   const sanitizedOperatingHours = hours.replace(/\s/g, '');
 
   // Escape early if it is 'Sunrise - Sunset'.
-  if (
+  return (
```
Comment (vbahinwillit) [inline src/applications/vaos/services/location/transformers.js:56]: Fix eslint warning

File: src/applications/vaos/services/location/transformers.js [hunk 10]
```diff
@@ -144,21 +140,6 @@ function transformOperatingHours(facilityHours) {
...
- * @returns A facility id with either 442 or 552 replaced with 983 or 984
- */
-function getTestFacilityId(facilityId) {
-  if (facilityId && (!environment.isProduction() || window.Cypress)) {
-    return facilityId.replace('442', '983').replace('552', '984');
-  }
-
-  return facilityId;
-}
-
```
Comment (vbahinwillit) [inline src/applications/vaos/services/location/transformers.js:161]: Moved to fix eslint cyclic dependency warning.

File: src/applications/vaos/services/vaos/index.js [hunk 11]
```diff
@@ -123,56 +122,6 @@ export function getAvailableV2Slots(facilityId, clinicId, startDate, endDate) {
...
-          useAcheron,
-        });
-        batch.push(p1);
-        return Promise.resolve([...batch].flat());
-      }, []);
-    }
-
-    return promise;
-  };
-})(3);
```
Comment (vbahinwillit) [inline src/applications/vaos/services/vaos/index.js:175]: Moved to fix eslint cyclic dependency warning.

**Review by ryanshaw — APPROVED** [review]
Looks good. Tested the following scenarios locally:

- Feature flag on, facility changed to 442: displayed links corrrectly
- Feature flag off, facility changed to 442: did not display links, correct behavior
- Feature flag on, facility not changed (remained 983): did not display links, correct behavior
- Feature flag ff, facility not changed (remained 983): did not display links, correct behavior

**Pull request merged**